android/MobileSdk/src/main/java/com/spruceid/mobile/sdk/IsoMdlPresentation.kt

```diff
@@ -28,29 +31,117 @@ class IsoMdlPresentation(
     val callback: BLESessionStateDelegate,
     val context: Context
 ) {
-    val uuid: UUID = UUID.randomUUID()
+    private val uuidCentral: UUID = UUID.randomUUID()
+    private val uuidPeripheral: UUID = UUID.randomUUID()
+
     var session: MdlPresentationSession? = null
     var itemsRequests: List<ItemsRequest> = listOf()
-    var bleManager: Transport? = null
+
+    // Dual transport instances for simultaneous operation
+    private var centralTransport: Transport? = null
+    private var peripheralTransport: Transport? = null
+
+    // Track which mode successfully connected first
+    @Volatile
+    private var connectedMode: String? = null
+    private val connectionLock = Any()
 
     @RequiresPermission(Manifest.permission.BLUETOOTH_CONNECT)
     fun initialize() {
         try {
-            session = initializeMdlPresentationFromBytes(this.mdoc, uuid.toString())
-            this.bleManager = Transport(this.bluetoothManager, context)
-            this.bleManager!!
-                .initialize(
-                    "Holder",
-                    this.uuid,
-                    "BLE",
-                    "Central",
-                    session!!.getBleIdent(),
-                    ::updateRequestData,
-                    callback
-                )
+            // Per ISO 18013-5: Always advertise both modes in QR code
+            Log.d("IsoMdlPresentation", "Initializing dual-mode presentation with separate UUIDs")
+            session = initializeMdlPresentationFromBytes(
+                this.mdoc,
+                CentralClientDetails(uuidCentral.toString()),
+                PeripheralServerDetails(uuidPeripheral.toString(), null)
+            )
+
+            // Display QR code immediately
             this.callback.update(mapOf(Pair("engagingQRCode", session!!.getQrCodeUri())))
+
+            // Start both transports simultaneously
+            startPeripheralTransport()
+            startCentralTransport()
```

Comment:
I think we should probably have some configuration for this, so the API consumer can pick which (or both) mode they want to use.